service/pure/src/main.rs

```diff
@@ -81,58 +84,80 @@ pub fn invoke() -> String {
 pub fn put(file_path: String) -> IpfsResult {
     log::info!("put called with {:?}", file_path);
     let timeout = load_config().timeout;
-    ipfs_put(file_path, timeout)
+    ipfs_put(file_path, load_config().local_multiaddr, timeout)
 }
 
 #[marine]
 pub fn get_from(hash: String, multiaddr: String) -> IpfsResult {
     log::info!("get called with hash: {}", hash);
-    let timeout = load_config().timeout;
+    let config = load_config();
+    let timeout = config.timeout;
+    let local_maddr = config.local_multiaddr;
 
     let particle_id = marine_rs_sdk::get_call_parameters().particle_id;
-    let connect_result = ipfs_connect(multiaddr, timeout.clone());
+    let connect_result = ipfs_connect(multiaddr, local_maddr.clone(), timeout.clone());
 
     if !connect_result.success {
         return connect_result;
     }
 
     let particle_vault_path = format!("/tmp/vault/{}", particle_id);
     let file_path = format!("{}/{}", particle_vault_path, hash);
-    let get_result = ipfs_get(hash, file_path.clone(), timeout);
+    let get_result = ipfs_get(hash, file_path.clone(), local_maddr, timeout);
 
     return if get_result.success {
         IpfsResult { success: true, result: file_path }
     } else {
         get_result
-    }
+    };
 }
 
 #[marine]
-pub fn get_multiaddr() -> IpfsResult {
-    load_multiaddr().into()
+pub fn get_external_multiaddr() -> IpfsResult {
+    load_external_multiaddr().into()
 }
 
 #[marine]
-pub fn set_multiaddr(multiaddr: String) -> IpfsResult {
+pub fn set_external_multiaddr(multiaddr: String) -> IpfsResult {
     let call_parameters = marine_rs_sdk::get_call_parameters();
-    if load_multiaddr().is_ok() || call_parameters.init_peer_id != call_parameters.service_creator_peer_id {
-        return eyre::Result::<()>::Err(eyre::eyre!("only service creator can set multiaddr only once")).into();
+    if load_external_multiaddr().is_ok() || call_parameters.init_peer_id != call_parameters.service_creator_peer_id {
+        return eyre::Result::<()>::Err(eyre::eyre!("only service creator can set external multiaddr only once")).into();
```

Comment:
Let's separate this error into two? `only creator can set only once` sounds a bit ridiculous